scripts/actions/spells/trust/tenzen_ii.lua

```diff
@@ -13,6 +13,16 @@ end
 
 spellObject.onMobSpawn = function(mob)
     xi.trust.message(mob, xi.trust.messageOffset.SPAWN)
+	
+	mob:addSimpleGambit(ai.t.TARGET, ai.c.ALWAYS, 0, ai.r.RATTACK, 0, 0)
+
+    mob:setAutoAttackEnabled(true)
+	
+	mob:addMod(xi.mod.STORETP, 95)
```

Comment:
His Tp per shot changes at exactly the levels that a samurai woudl gain the next store tp Trait, hoever there is no delay value that is consistent with his TP gains when combined with Store TP. Somebody at SE made it up. This ironically penalizes his TP gains before the final tier of STP, because Semih Lefana will get that exact same value at lower levels. Its like they punished him for being a Samurai.

Reply:
Comparing both Tenzen II scripts, they look nothing like each other.
yours -> them
Auto attack true -> Auto attack false
Trust distance mid range -> Trust distance long range
TP opener random -> TP opener Highest

I have to ask, did you get the info from retail?
Couse some1has to be wrong here

Reply:
He and Semi also have much shorter delays than their TP gains would indicate, btw. Our implementation of Joachim's throw is also wrong, which i think was the basis for carver's Tenzen II script.

tl;dr ranged attacker trusts need some special handling and a total do-over

Reply:
> Only uses his weaponskill to open skillchains when another party member has 1000 TP. Will wait at 3000 TP indefinitely if no other party members gain TP.

Reply:
BG doesn't specify what rules for -who- he opens for, just that somebody has to be > 1000

Reply:
> I was working off carver's and comparing with my live retail account. H e was mostly behaving right but teh TOP gains per ranged shot were just wrong.
> 
> To get His TP gains retail accurate I basically need some kind of override to set a specific amount of TP per ranged attack based on his level: it changes at the exact levels he would gain the store TP trait, but retail testing (BY ME) shows there is no valid delay rating that would result in correct amount of TP when factoring in storeTP. Bets bet would be zero his TP gains and then literally just do a conditional for his level and call `addTP()`

I'll look into what I can do specifically for this without affecting other ranged attackers.

Reply:
sorry for multi-posting:

presumably, as soon as somebody hits 1k he's gonna go immediately, but we can double check that easy enough

Reply:
> I'll look into what I can do specifically for this without affecting other ranged attackers.

if you can manage getting him retail accurate, awesome. if not just don't even try to fix his TP gain for now, leave out STP mods in script he already would be getting traits from job  and SE did something crazy with trust TP gains on ranged attackers in general - none of them match their actual delays as seen in game. they all get far more tp than their delay indicated.

Reply:
at least Tenzen is the only one with variable TP gains. If he were real he'd probably be opening support tickets asking why he was being nerfed for no reason and wishing he leveled war instead.

Reply:
The short answer would be, make him not samurai but that's not the right answer lol